python_modules/dagster-graphql/dagster_graphql/implementation/utils.py

```diff
@@ -117,18 +117,25 @@ def watch(fn: Callable[[Exception], None]) -> Iterator[None]:
         finally:
             ErrorCapture.observer.reset(token)
 
+    capturing_error = False
+
 
 def capture_error(
     fn: Callable[P, T]
 ) -> Callable[P, Union[T, "GrapheneError", "GraphenePythonError"]]:
     def _fn(*args: P.args, **kwargs: P.kwargs) -> T:
         try:
+            if ErrorCapture.capturing_error:
+                raise Exception("capture_error invoked more than once within a single graphql call")
```

Comment:
another option would be skipping the middleware if its already been enabled? maybe with a warning?